### PR TITLE
Meir/feat/homepage

### PR DESCRIPTION
**Pull request by MeirJC**

### CODE REVIEW

File: src/pages/homepage/HomePage.Style.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import styled from "styled-components";
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.Style.js]: import styled from 'styled-components/macro'
its easier for debugging purposes

File: src/pages/homepage/HomePage.Style.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  margin: 10rem auto 12rem;
+`;
+export const HeaderPage = styled.header`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  margin: 2rem auto;
+`;
+
+export const H1 = styled.h1`
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.Style.js]: i suggest to give it another name than HTML tags , the naming is depending on the usage if its title -> Title , SubTitle->SubTitle

File: src/pages/homepage/HomePage.Style.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  font-size: 5rem;
+  letter-spacing: -0.425rem;
+  font-weight: 700;
+  font-family: "kopik", sans-serif;
+  margin: 0;
+  padding: 0;
+`;
+
+export const H3Tag = styled.h3`
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.Style.js]: also here

File: src/pages/homepage/HomePage.Style.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  cursor: pointer;
+`;
+
+export const UserDetails = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+export const PTag = styled.p`
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.Style.js]: also here

File: src/pages/homepage/HomePage.Style.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  font-size: 1.2rem;
+  font-weight: 500;
+  &:placeholder-shown {
+    text-align: right;
+  }
+  &:focus {
+    outline: none;
+  }
+`;
+export const A = styled.a`
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.Style.js]: also here

File: src/pages/homepage/Global.Style.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
+import { createGlobalStyle } from "styled-components";
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/Global.Style.js]: use this component in  main component (app.js)

File: src/pages/homepage/Global.Style.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+const GlobalStyle = createGlobalStyle`
+  
+@font-face {
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/Global.Style.js]: Separate font from html style and import it in index.css

File: src/pages/homepage/HomePage.js [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+  const [signUp, setSignUp] = useState(false);
+  //? State for the join mail-list checkbox
+  const [mailSign, setMailSign] = useState(false);
+  //? State for the valid email
+  const [isValidMail, setIsValidMail] = useState(true);
+  //? Show / Hide why we ask for the email
+  const [showWhyMail, setShowWhyMail] = useState(false);
+  //? Importing the context
+  const { role, setRole, dummyUser, isAuthenticated } = useAuthContext();
+  // console.log("useAuthContext: ", useAuthContext());
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.js:25]: remove console logs

File: src/pages/homepage/HomePage.js [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+      username,
+      "password: ",
+      password,
+      "signUp: ",
+      signUp,
+      "mailSign: ",
+      mailSign
+    );
+    //? Chaking if the username is a valid email
+    if (!username.match(/[^\s@]+@[^\s@]+\.[^\s@]+/gi)) {
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.js]: username -> userEmail

Comment (AmeerLalaDeveloper) [reply src/pages/homepage/HomePage.js]: put validation in utils file

File: src/pages/homepage/HomePage.js [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+      console.log("EMAIL username: NOT VALID EMAIL! ", username);
+      setIsValidMail(false);
+    } else if (username.match(/[^\s@]+@[^\s@]+\.[^\s@]+/gi)) {
+      console.log("EMAIL username: VALID! ", username);
+      setIsValidMail(true);
+    }
+    //? Checking if the username (email) and the password matches
+    if (username === dummyUser.email && password === dummyUser.password) {
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.js]: when we connect to airtable , use this validation in the backend

File: src/pages/homepage/HomePage.js [hunk 11]
```diff
@@ -0,0 +1,194 @@
...
+    } else {
+      console.log(
+        "username and paswword are NOT matching, authentication FAILD"
+      );
+    }
+  };
+  return (
+    <div>
+      <GlobalStyle />
+      {!role ? <Header.Main /> : <Header.Page />}
```
Comment (AmeerLalaDeveloper) [inline src/pages/homepage/HomePage.js]: reuse Header Style with different props

File: src/context/authContext.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    dummyUser,
+  };
+
+  return (
+    <AuthContext.Provider value={valueToShare}>{children}</AuthContext.Provider>
+  );
+}
+
+export { Provider };
+export default AuthContext;
```
Comment (AmeerLalaDeveloper) [inline src/context/authContext.js:41]: export your custom hook here

File: src/pages/homepage/Global.Style.js (hunk 6 above)
Comment (MeirJC) [reply src/pages/homepage/Global.Style.js]: eventually removed global styles

File: src/pages/homepage/Global.Style.js (hunk 7 above)
Comment (MeirJC) [reply src/pages/homepage/Global.Style.js]: eventually removed global styles

File: src/pages/homepage/HomePage.js (hunk 11 above)
Comment (MeirJC) [reply src/pages/homepage/HomePage.js]: handled with props

**Pull request merged**